FET add retry mechanism for fetch_xx functions (#21691)
scikit-learn/scikit-learn: doc/whats_new/v1.2.rst

  Cython implementation, providing 2-4x speedups.
  :pr:`23127` by :user:`Meekail Zain <micky774>`

- |Enhancement| Adds optional arguments `n_retries` and `delay` to functions 
  :func:`datasets.fetch_20newsgroups`,
  :func:`datasets.fetch_20newsgroups_vectorized`,
  :func:`datasets.fetch_california_housing`,
  :func:`datasets.fetch_covtype`,
  :func:`datasets.fetch_kddcup99`,
  :func:`datasets.fetch_lfw_pairs`,
  :func:`datasets.fetch_lfw_people`,
  :func:`datasets.fetch_olivetti_faces`,
  and :func:`datasets.fetch_species_distributions`.

>   By default, all those functions will retry 3 times by default in case of network failure.

Reviewer: Perhaps this makes it a bit cleaner.

```suggestion
  By default, the functions will retry up to 3 times in case of network failures.
```
